src/metpy/plots/patheffects.py

```diff
@@ -0,0 +1,245 @@
+#  Copyright (c) 2020,2022 MetPy Developers.
+#  Distributed under the terms of the BSD 3-Clause License.
+#  SPDX-License-Identifier: BSD-3-Clause
+"""Add effects to matplotlib paths."""
+from functools import cached_property
+import itertools
+
+import matplotlib.colors as mcolors
+import matplotlib.path as mpath
+import matplotlib.patheffects as mpatheffects
+import matplotlib.transforms as mtransforms
+import numpy as np
+
+from ..package_tools import Exporter
+
+exporter = Exporter(globals())
+
+
+class Front(mpatheffects.AbstractPathEffect):
+    """Provide base functionality for plotting fronts as a patheffect.
+
+    These are plotted as symbol markers tangent to the path.
+
+    """
+
+    symbol = mpath.Path([[0, 0], [0, 1], [1, 1], [1, 0], [0, 0]],
```

Comment:
<img width="512" alt="image" src="https://user-images.githubusercontent.com/24628426/230658889-376d593b-eb4c-4908-92ba-632ac4770545.png">

Worth excluding this attribute from autodoc? (here specifically from `WarmFront`) Or otherwise hiding it away? I suppose we can leave it pseudo-visible if we want to show people how they can draw custom symbols at some point.

Reply:
Yeah, let's make `_symbol` private since I don't really think people need to get to this.